### ISSUE TITLE
docs: clarify the contribution process

### ISSUE THREAD

**Issue by petermakowski** [canonical/vanilla-framework#4905]
As a developer, I want to have clear guidelines when a component/new styles should be upstreamed to vanilla, and what's the process in detail. What are the criteria for adding such component?

We should clarify the contribution process, including the fact that we want to err on the side of upstreaming new components.

**Comment by bartaz** [comment]
Yes, I think we can make https://github.com/canonical/vanilla-framework/blob/main/CONTRIBUTING.md a bit more clear in this regard. Currently it focuses on how to run a project, but the section on "reporting bugs" should be extended to clarify how to propose new components or amendments to them.

**Comment by bartaz** [comment]
Triage: we are in the process of discussing design system definitions and processes, and contribution process is part of that. Once we have the decisions, the documentation will be adjusted accordingly.
